### PR TITLE
CODENVY-673: Fix IndexOutOfBoundsException

### PR DESCRIPTION
**Pull request by vinokurig**
Sometimes `IndexOutOfBoundsException` appears when starting workspace.
@garagatyi @mmorhun please review

### CODE REVIEW

File: plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/RemoteDockerNode.java [hunk 1]
```diff
@@ -104,14 +104,19 @@ public void bindWorkspace() throws MachineException {
                                                                          .withDetach(false));
             final List<String> ownerIds = new ArrayList<>(4);
             final ValueHolder<String> error = new ValueHolder<>();
-            dockerConnector.startExec(StartExecParams.create(exec.getId()), message -> {
-                if (message.getType() == LogMessage.Type.STDOUT) {
-                    ownerIds.add(message.getContent());
+            String errorMessage = "Can't detect container user ids to chown backed up workspace " + workspaceId + " files.";
```
Comment (garagatyi) [inline plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/RemoteDockerNode.java]: Please move to field to reuse memory.

File: plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/RemoteDockerNode.java [hunk 2]
```diff
@@ -104,14 +104,19 @@ public void bindWorkspace() throws MachineException {
...
+            dockerConnector.startExec(StartExecParams.create(exec.getId()), logMessage -> {
+                if (logMessage.getType() == LogMessage.Type.STDOUT) {
+                    ownerIds.add(logMessage.getContent());
                 } else {
-                    LOG.error("Can't detect container user ids to chown backed up workspace " + workspaceId + "files. " + message);
-                    error.set("Can't detect container user ids to chown backed up workspace " + workspaceId + "files");
+                    LOG.error(errorMessage + " " + logMessage);
+                    error.set(errorMessage);
```
Comment (garagatyi) [inline plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/RemoteDockerNode.java]: If multiple logs comes from STDERR then error will be rewritter multiple times

File: plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/RemoteDockerNode.java [hunk 3]
```diff
@@ -104,14 +104,19 @@ public void bindWorkspace() throws MachineException {
...
                 }
             });
+            if (ownerIds.size() < 2 && error.get() == null) {
+                LOG.error(errorMessage  + (ownerIds.size() == 1 ? " " + ownerIds.get(0) : ""));
+                error.set(errorMessage);
```
Comment (garagatyi) [inline plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/RemoteDockerNode.java]: This error in not informative. Please add explanation to error what has happened.

**Comment by vinokurig** [comment]
reworked the logic of throwing error

**Comment by mmorhun** [comment]
+1

**Comment by garagatyi** [comment]
LGTM

**Pull request merged**